### PR TITLE
Stop the UIController properly in the GitHubPane

### PR DESCRIPTION
**Pull request by shana**
Make sure existing UI is stopped whenever the following conditions change:

- [x] The active repository changes
- [x] The user login status changes

Also make sure `Reload` is reentrant, all those await calls means it can get called multiple times.

### CODE REVIEW

**Comment by shana** [comment]
Still have an annoying problem with the list sometimes not refreshing, even though it clearly changes. But otherwise, looking nice

pr-list-view2

File: src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs [hunk 1]
```diff
@@ -19,13 +19,16 @@
 using ReactiveUI;
 using System.Collections.Generic;
 using System.Threading.Tasks;
+using System.Diagnostics;
```
Comment (paladique) [inline src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs]: Looks like this isn't being used.

File: src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs [hunk 2]
```diff
@@ -105,53 +108,103 @@ protected override void RepoChanged(bool changed)
...
+                var isLoggedIn = await connection.IsLoggedIn(hosts);
+                if (reloadCallId != latestReloadCallId)
+                    return;
+
+                IsLoggedIn = isLoggedIn;
             }
+
+            if (!IsGitHubRepo.Value)
+            {
+                //LoadView(UIViewType.NotAGitHubRepo);
```
Comment (paladique) [inline src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs]: Doesn't look like this if statement is doing anything anymore, I uncommented to investigate and it looks like it's broken, do we intend to add `NotAGitHubRepo` to `UIViewType` at some point in the future?

Comment (shana) [inline src/GitHub.VisualStudio/UI/Views/GitHubPaneViewModel.cs]: Yeah, it's a placeholder for @grokys to stuff things into 😄

**Comment by paladique** [comment]
This looks great :+1:

**Pull request merged**